controllers/config_controller.go

```diff
@@ -67,21 +66,12 @@ func (r *Layer2NetworkConfigurationReconciler) Reconcile(ctx context.Context, _
 }
 
 // SetupWithManager sets up the controller with the Manager.
-func (r *Layer2NetworkConfigurationReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	// Create empty request for changes to node
-	nodesMapFn := handler.EnqueueRequestsFromMapFunc(func(_ context.Context, _ client.Object) []reconcile.Request { return []reconcile.Request{{}} })
-	nodePredicates := predicate.Funcs{
-		CreateFunc: func(_ event.CreateEvent) bool { return false },
-		UpdateFunc: func(e event.UpdateEvent) bool {
-			return os.Getenv(healthcheck.NodenameEnv) == e.ObjectNew.GetName() && !cmp.Equal(e.ObjectNew.GetLabels(), e.ObjectOld.GetLabels())
-		},
-		DeleteFunc:  func(_ event.DeleteEvent) bool { return false },
-		GenericFunc: func(_ event.GenericEvent) bool { return false },
-	}
-
+func (r *ConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	err := ctrl.NewControllerManagedBy(mgr).
-		For(&networkv1alpha1.Layer2NetworkConfiguration{}).
-		Watches(&corev1.Node{}, nodesMapFn, builder.WithPredicates(nodePredicates)).
+		Named("config controller").
+		Watches(&networkv1alpha1.Layer2NetworkConfiguration{}, &handler.EnqueueRequestForObject{}).
+		Watches(&networkv1alpha1.RoutingTable{}, &handler.EnqueueRequestForObject{}).
+		Watches(&networkv1alpha1.VRFRouteConfiguration{}, &handler.EnqueueRequestForObject{}).
```

Comment:
I've added a handler using `handler.EnqueueRequestsFromMapFunc`. It will always return an 1 element array of empty `ctrl.Request` so I think those should be batched by controller-runtime.